src/test/java/ch/uzh/ifi/hase/soprafs23/controller/APIControllerTest.java

```diff
@@ -0,0 +1,156 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Import(RestTemplateConfig.class)
+public class APIControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private GroupService groupService;
+
+    @MockBean
+    private UserService userService;
+
+    @MockBean
+    private APIService apiService;
+    @Mock
+    private RestTemplate restTemplate;
+    @InjectMocks
+    private APIController apiController;
+
+    private Group testGroup;
+    private User testUser;
+    private Recipe testRecipe;
+
+    @BeforeEach
+    public void setUp() {
+        testGroup = new Group();
+        testGroup.setId(1L);
+        testGroup.setHostId(1L);
+
+        testUser = new User();
+        testUser.setId(1L);
+
+        testRecipe = new Recipe();
+        testRecipe.setId(1);
+        testRecipe.setTitle("Test Recipe");
+        testRecipe.setReadyInMinutes(30);
+        testRecipe.setPricePerServing(5.00);
+
+        doReturn(testGroup).when(groupService).getGroupById(anyLong());
+        doReturn(testUser).when(userService).getUserById(anyLong());
+        doReturn(1L).when(userService).getUseridByToken("valid-token");
+        doReturn(testRecipe).when(apiService).getHostRecipe(testUser);
+    }
+
+    @Test
+    public void getRandomRecipe_success() throws Exception {
+        mockMvc.perform(get("/groups/{groupId}/result", 1L)
+                        .header("X-Token", "valid-token")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+    @Test
+    public void getHostRecipe_returnsConflictStatusCode_whenNoRecipesFound() {
```

Comment:
could you pls add the error codes either within the test or next to it as comments? Helps reading the code :)